mifosng-provider/src/main/java/org/mifosplatform/portfolio/loanaccount/service/LoanWritePlatformServiceJpaRepositoryImpl.java

```diff
@@ -2890,9 +2890,9 @@ public void recalculateInterest(final long loanId) {
     }
 
     @Override
-    public CommandProcessingResult recoverFromGuarantor(final Long loanId) {
+    public CommandProcessingResult recoverFromGuarantor(final Long loanId, LocalDate recoverFromGuarantor) {
```

Comment:
should pass JsonCommand  to this methd insteadof recoverFromGuarantor date and add validation for date.  
also pass jsoncommand from RecoverFromGuarantorCommandHandler.java